test/Test/Oscoin/Protocol.hs

```diff
@@ -0,0 +1,102 @@
+module Test.Oscoin.Protocol
+    ( tests
+    ) where
+
+import           Oscoin.Prelude
+
+import qualified Oscoin.Consensus.Config as Consensus
+import           Oscoin.Consensus.Nakamoto (blockScore)
+import           Oscoin.Crypto.Blockchain
+import           Oscoin.Protocol
+import           Oscoin.Storage.Block.Abstract as Abstract
+import qualified Oscoin.Storage.Block.Memory as Memory
+import           Oscoin.Storage.Block.Orphanage
+import qualified Oscoin.Storage.Block.STM as STM
+import qualified Oscoin.Time.Chrono as Chrono
+
+import           Oscoin.Test.Crypto
+import           Oscoin.Test.Crypto.Blockchain.Generators
+                 (ForkParams(..), genBlockchainFrom, genOrphanChainsFrom)
+import           Oscoin.Test.Storage.Block.SQLite (DummySeal, defaultGenesis)
+import           Oscoin.Test.Util (Condensed(..))
+
+import           Control.Monad.State (modify')
+import           Data.ByteArray.Orphans ()
+
+import           Test.Oscoin.DummyLedger
+import           Test.QuickCheck.Extended
+import           Test.QuickCheck.Monadic
+import           Test.Tasty
+import           Test.Tasty.QuickCheck hiding ((===))
+
+tests :: Dict (IsCrypto c) -> TestTree
+tests d = testGroup "Test.Oscoin.Protocol"
+    [ testProperty "prop_StmBlockStoreEquivalence"  (prop_StmBlockStoreEquivalence d)
```

Comment:
I liked the old name a bit better -- or, at least, I would suggest making this one slightly more descriptive, as it's a bit obscure now.